Source/ablastr/parallelization/MPIInitHelpers.cpp

```diff
@@ -32,18 +32,18 @@ namespace ablastr::parallelization
     int
     mpi_thread_required ()
     {
-        int thread_required = -1;
 #ifdef AMREX_USE_MPI
-        thread_required = MPI_THREAD_SINGLE;  // equiv. to MPI_Init
+        return MPI_THREAD_SINGLE;  // equiv. to MPI_Init
```

Comment:
This is wrong. The previous logic intentionally overwrites `thread_required` a few times.

Example path that now returns `MPI_THREAD_SINGLE` instead of `MPI_THREAD_FUNNELED`
- MPI is ON and OMP is used but `AMREX_MPI_THREAD_MULTIPLE` is not available.

Example path that now returns `MPI_THREAD_SINGLE` instead of `MPI_THREAD_MULTIPLE`
- MPI is ON and `AMREX_MPI_THREAD_MULTIPLE` is available.

Reply:
Which test forced this? Looks unnecessary to me. The compiler knows how to optimize out multiple assignments.

Reply:
I am sorry, @ax3l . I should have marked this PR as `[WIP]`.
You are totally right, this is an error. I've temporarily reverted the code to the previous version.

Reply:
The check that forces some changes is `clang-analyzer-deadcode.DeadStores`